lib/compute/agent-node-config.ts

```diff
@@ -48,12 +49,24 @@ export class AgentNodeConfig {
     this.STACKREGION = stack.region;
     this.ACCOUNT = stack.account;
 
-    const key = new KeyPair(stack, 'AgentNode-KeyPair', {
-      name: 'AgentNodeKeyPair',
-      description: 'KeyPair used by Jenkins Main Node to SSH into Agent Nodes',
+    const key = new CfnKeyPair(stack, 'AgentNode-KeyPair', {
+      keyName: 'AgentNodeKeyPair',
+      tags: [
+        {
+          key: 'jenkins:credentials:type',
+          value: 'sshUserPrivateKey',
+        },
+      ],
     });
-    Tags.of(key)
-      .add('jenkins:credentials:type', 'sshUserPrivateKey');
+
+    // Create a secret for storing the actaul value of key-pair so that jenkins secret-manager plugin is able to fetch it
+    const keyPairSecret = new Secret(stack, 'AgentNodeKeyPair', {
+      secretName: 'jenkins-agent-node-key-pair',
+      description: 'ssh key for jenkins main node to connect to agent nodes',
+    });
+    Tags.of(keyPairSecret).add('jenkins:credentials:type', 'sshUserPrivateKey');
```

Comment:
So we need this tag for both secret as well as actual ec2 key pair?

Reply:
We have to add this tag so that jenkins can read this and use it as ssh key, isnt it?

Reply:
This tag is mandatory for the secret but not for the actual key-pair. 
Don't see any issue with adding the same tag to key-pair as well.